libvmi/read.c

```diff
@@ -97,13 +97,13 @@ vmi_read(
 
         if(dtb) {
             paddr = vmi_pagetable_lookup(vmi, dtb, start_addr + buf_offset);
+            if (!paddr) {
+                return buf_offset;
+            }
         } else {
             paddr = start_addr + buf_offset;
         }
 
-        if (!paddr) {
```

Comment:
On the other hand we have been restricting access to paddr 0 even wtih physical access directly so this is clearly an issue that needs to be fixed either way.